sqlglot/generator.py

```diff
@@ -2322,6 +2324,33 @@ def tochar_sql(self, expression: exp.ToChar) -> str:
 
         return self.sql(exp.cast(expression.this, "text"))
 
+    def dictproperty_sql(self, expression: exp.DictProperty) -> str:
+        this = self.sql(expression, "this")
+        kind = self.sql(expression, "kind")
+        settings = []
+        for s in expression.args["settings"]:
```

Comment:
settings is not required in your spec, so this could be not set, do expression.args("settings") or []:

Reply:
but givven how you're using it, you should really do

self.expressions(expression, key="settings") and make sure the key value is generated correctly

Reply:
I'm not sure how to use it with `self.expressions()` as these are `key value` and not `key=value`

Reply:
@tobymao unfortunately only the dictionary `kind` properties are `key value` in CH
So these are pretty specific. I can probably subclass `exp.Property` for that, but is it worth it?

Reply:
yes let's do that

Reply:
Done!